frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerThread.java

```diff
@@ -94,16 +96,30 @@ public void run() {
         BaseModelRequest req = null;
         try {
             while (running.get()) {
-                req = aggregator.getRequest();
+                req = aggregator.getRequest(currentThread.getId());
                 backendChannel.write(req);
                 backendChannel.flush();
 
-                ModelWorkerResponse reply = replies.take();
-                aggregator.sendResponse(reply);
+                // TODO: Change this to configurable param
+                ModelWorkerResponse reply = replies.poll(120, TimeUnit.SECONDS);
+
+                if (reply != null) {
+                    aggregator.sendResponse(reply);
+                } else if (req.getIsSynchronous()) {
+                    throw new WorkerInitializationException(
+                            "Backend worker did not respond in given time");
+                }
+
+                if (req.getCommand().equals(CommandStrings.PREDICT)) {
+                    model.resetFailedInfReqs();
```

Comment:
What's the failedInReqs used for?

Reply:
This is to see how many subsequent inference requests failed. This can be a metric generated by the front end to show how many subsequent inference requests failed.